src/Templates/ModelTemplate.cshtml

```diff
@@ -223,7 +223,7 @@ else
     </text>
 }
 
-@if (Model.HasPolymorphicFields || Model.HasFlattenedFields)
+@if (Model.HasPolymorphicFields || Model.HasFlattenedFields || Model.AdditionalPropertiesField != null)
```

Comment:
no regressions because of this ? also nit maybe extract in a needsCustomMarshalling method

Reply:
I still have to test that part, I will report back when I have those results.

Reply:
great, well approved if everything is fine feel free to merge